src/main/java/org/jabref/logic/importer/fileformat/PdfMergeMetadataImporter.java

```diff
@@ -125,14 +125,13 @@ private List<BibEntry> fetchIdsOfCandidates(List<BibEntry> candidates) {
 
             Optional<String> eprint = candidate.getField(StandardField.EPRINT);
             if (eprint.isPresent()) {
-                // It's not exactly right, arXiv is not the only preprint service.
-                // There should be check for `archivePrefix = {arXiv}`, but I'm worried if user didn't set it.
+                // This code assumes that any `eprint` refers to an arXiv preprint, which is not correct.
```

Comment:
I liked the old comment better, because it provided some hints to fix this (prefix of the Id) -- Maybe you can point to some other JabRef class for Id parsing or arxivIdClean up